### PR TITLE
XOL-4143 MailChimp plugin

### PR DESCRIPTION
**Pull request by jeetsang**

### CODE REVIEW

File: Logger.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import winston from 'winston';
+
+const Logger = new winston.Logger({
+    level: 'info',
+    json: true,
+});
+Logger.add(winston.transports.Console);
+Logger.stream = {
```
Comment (srishaharidas) [inline Logger.js]: what about logentries?

File: app.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+app.use(express.static(path.join(__dirname, 'public')));
+app.use(morgan('combined', {stream: Logger.stream}));
+
+app.use('/', router);
+
+// catch 404 and forward to error handler
+app.use(function(req, res, next) {
+    next(createError(404));
+});
+
```
Comment (srishaharidas) [inline app.js]: you should have an error handler to catch all errors in one place and include it here

File: config/default.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+    "appStore": {
+        "url": "https://elrond.xola.com"
```
Comment (srishaharidas) [inline config/default.json]: default values should not be production urls

File: controllers/OptionController.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import MailChimpService from '../services/MailChimpService';
+import AppStoreService from '../services/AppStoreService';
+import {MAIL_CHIMP_API_KEY_NAME, YES, NO} from '../constants/constants';
+
+const OptionController = {
```
Comment (srishaharidas) [inline controllers/OptionController.js]: is this supposed to `OptinController`?

File: controllers/OptionController.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    lists: async (req, res) => {
```
Comment (srishaharidas) [inline controllers/OptionController.js]: consider renaming this to `getLists` to make it more readable

File: controllers/WebhookController.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import MailChimpService from '../services/MailChimpService';
+import AppStoreService from '../services/AppStoreService';
+
+const WebhookController = {
+    processWebhook: async (req, res) => {
+        const {eventName, data} = req.body;
+        switch (eventName) {
+            case 'traveler.create':
+            case 'traveler.update':
+                let config = await AppStoreService.getInstallationConfig(data.traveler.seller.id);
```
Comment (srishaharidas) [inline controllers/WebhookController.js]: wont you already have the installation config in the webhook payload itself?

File: manifest/register_app.json [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        }
+      ]
+    },
+    {
+      "title": "MailChimp Lists",
+      "fields": [
+        {
+          "type": "select",
+          "name": "mail_chimp_list",
+          "label": "Select list",
```
Comment (srishaharidas) [inline manifest/register_app.json]: sentence case. `Select List`

File: Logger.js (hunk 1 above)
Comment (jeetsang) [reply Logger.js]: We don't need this and have confirmed with Anush.

File: app.js (hunk 2 above)
Comment (jeetsang) [reply app.js]: No need of this express has a default error handler which will take care of it.
https://expressjs.com/en/guide/error-handling.html

File: config/default.json (hunk 3 above)
Comment (jeetsang) [reply config/default.json]: it is going to be private plugin. Therefore is it alright  to have it here.

File: controllers/OptionController.js (hunk 4 above)
Comment (jeetsang) [reply controllers/OptionController.js]: No, It is actually a configuration option provider to store-ui  for mailchip list and opt-in config.
I could rename this to configuration controller.

File: controllers/OptionController.js (hunk 5 above)
Comment (jeetsang) [reply controllers/OptionController.js]: done

File: controllers/WebhookController.js (hunk 6 above)
Comment (jeetsang) [reply controllers/WebhookController.js]: Here, plugin is only listening to xola events (traveler.create and traveler.update) not to installation events of xola.Therefore it has to fetch installation config.

File: manifest/register_app.json (hunk 7 above)
Comment (jeetsang) [reply manifest/register_app.json]: done

File: app.js (hunk 2 above)
Comment (srishaharidas) [reply app.js]: dont think this will this catch unexpected errors though

File: config/default.json (hunk 3 above)
Comment (srishaharidas) [reply config/default.json]: https://elrond.xola.com is still never your default app store url irrespective of this being a private plugin

**Pull request merged**